### PR TITLE
perf: Remove huge EventUser query from GroupTagValueSerializer

### PR DESCRIPTION
**Pull request by bretthoerner**
GroupTagValueSerializer used to fetch *all* EventUser records for a
project (which could have been millions -- the reason we're making this
change is because it showed up in the top slowest query logs).

The values were used to build a lookup dict to retrieve the proper
'label'. It turns out the label is simply the raw value without the
prefix. (e.g. Tag Value: "id:10", Label: "10") So instead of doing the
query and building the dict, we just strip the prefix if one exists.

This should be functionally equivalent but without doing the query.

### CODE REVIEW

**Review by mattrobenolt — DISMISSED** [review]
If tests pass 🥇

**Comment by bretthoerner** [comment]
@dcramer can you verify whether some functionality is no longer needed?

This is breaking here: https://github.com/getsentry/sentry/blob/master/tests/sentry/api/serializers/test_grouptagvalue.py#L52

```
Traceback (most recent call last):
  File "/Users/brett/Development/sentry/sentry/tests/sentry/api/serializers/test_grouptagvalue.py", line 52, in test_with_no_tagvalue
    assert result['name'] == grouptagvalue.value
AssertionError: assert 'foo@example.com' == 'email:foo@example.com'
  - foo@example.com
  + email:foo@example.com
  ? ++++++
```

This test seems to be proving that *if* an `EventUser` record doesn't exist, the `GroupTagValue` won't be serialized with a `label` (e.g. the `email:` prefix won't be removed).

1. I'm not sure I understand the value in that?
2. If a `sentry:user` tag was used, wouldn't an `EventUser` necessarily exist?

In short: can I remove the test, or am I missing something important here?

**Comment by dcramer** [comment]
I don’t recall the reasoning so I’d say use your best judgement

**Review by mattrobenolt — APPROVED** [review]
